static/src/javascripts/projects/commercial/modules/header-bidding/slot-config.js

```diff
@@ -129,9 +129,21 @@ const getSlots = (contentType: string): Array<PrebidSlot> => {
     }
 };
 
-export const getPrebidAdSlots = (
+export const getHeaderBiddingAdSlots = (
     ad: Advert,
-    contentType: string
-): Array<PrebidSlot> => filterByAdvert(ad, getSlots(contentType));
+    slotFlatMap?: HeaderBiddingSlot => HeaderBiddingSlot[]
```

Comment:
This optional flatMap parameter is making it much more difficult for me to understand what is going on here. What's the difference between a call site that wants to use the `slotFlatMap` logic (which includes things like "fiddling with top-above-nav") and call sites that do not?

Reply:
for cases when the slotFlatMap is not set it will pass the below function on next line which will initialiaze it :
`const effectiveSlotFlatMap = slotFlatMap || (s => [s]); // default to identity`
essentially just taking the slot and returning it as list (all that logic was existent already)